### ISSUE TITLE
Fit to min of column header or contents

### ISSUE THREAD

**Issue by dpatel20** [olifolkerd/tabulator#321]
Hi -

I am using v3 BETA and wonder if there is a way to fit the column width to `min(headerTextWidth, contentTextWidth)`?

Right now, I can fit to content but that can mean the header text is lost.

PS: Is it me, or does colMinWidth not work in v3.0?

**Comment by olifolkerd** [comment]
Documentation  fail there, it has now been renamed to ***columnMinWidth*** to fit with the new naming standards.

I have updated the documentation.

Thanks for the feedback :)

**Comment by olifolkerd** [comment]
i closed that a bit to early, at the moment there is no way to make it so the header doesn't get truncated if the data is smaller, but i was just thinking that myself. i will have a look at adding it in the next couple of days

**Comment by olifolkerd** [comment]
Hey,

I have added an update to the fit to data functionality so it will include the header in its calculation so you should no longer end up with truncated header text :)

**Comment by dpatel20** [comment]
Sorry to resurrect this now - just had chance to test this. It works well. However, when I have a `headerFilter` it goes back to a standard width. Can this be included in the calc?

**Comment by olifolkerd** [comment]
I cant seem to replicate your issue, everything works fine when i use header filters.

What do you mean by "standard width"?

Could you post a copy of your Tabulator constructor.

Cheers

Oli

**Comment by dpatel20** [comment]
Hi Oli -

Without header filters, the column width=header text width.
With header filters, the column width=filter input box width. When I say standard width, I assume that the input box has some kind of default width?

```JS
$("#example-table-demo").tabulator({
    height: "75%",
    index: "objectId",
    //fitColumns:true,
    //columnMinWidth : 100,
    columns:[
      {title:"AGE GROUP", field:"AGE_GROUP", sorter:"alphanum", headerFilter:"input"},
      {title:"GROUP", field:"GROUP", sorter:"string", headerFilter:"input"},
      {title:"TYPE", field:"MATCH_TYPE", sorter:"string",headerFilter:"input"},
      {title:"TEAM1", field:"TEAM1", sorter:"string", headerFilter:true, editor:dropdownTeams},
      {title:"TEAM1SCORE", field:"TEAM1SCORE", sorter:"number", headerFilter:"number"},
      {title:"TEAM2", field:"TEAM2", sorter:"string", headerFilter:true, editor:dropdownTeams},
      {title:"TEAM2SCORE", field:"TEAM2SCORE", sorter:"number",headerFilter:"number"},
      {title:"PITCH", field:"PITCH", sorter:"number", headerFilter:"number", editor:"number"},
      {title:"ORDER", field:"ORDER", sorter:"number", headerFilter:"number", editor:"number"},
      {title:"ET", field:"ET", sorter:"string", headerFilter:"input"},
    ],
  });
```
A picture might help...notice how `GROUP` column is the width of the header title. But `TYPE` is much wider than the header title (and it is also wider than the content of the column). I think the filter input box is causing the column to much wider.

![image](https://user-images.githubusercontent.com/28842105/26926070-fb66b1be-4c43-11e7-8fe8-2d32c662deb9.png)

**Comment by edwulit** [comment]
@olifolkerd 
@dpatel20 

Hi, was a solution for the column widths expanding with header filter enabled ever found?

**Comment by olifolkerd** [comment]
Hey All,

you will be happy to hear i have just pushed a fix for this to the 3.4 branch which will be released tomorrow.

Cheers

Oli :)
